lib/driver.rb

```diff
@@ -0,0 +1,73 @@
+require_relative 'csv_record'
+
+module RideShare
+  class Driver < CsvRecord
+    attr_reader :name, :vin, :status, :trips
+
+    def initialize(id:, name:, vin:, status: :AVAILABLE, trips: nil)
+      super(id)
+
+      @name = name
+      @vin = vin
+      @status = status.to_sym
+      @trips = trips || []
+
+      if @vin.length != 17
+        raise ArgumentError, "vin: #{vin} is not valid"
+      end 
+      
+      if (@status == :AVAILABLE || @status == :UNAVAILABLE) == false 
+        raise ArgumentError, "status:#{status} is not a valid status"
+      end
+    end
+
+    def add_trip(trip)
+        @trips << trip
+    end 
+
+    def average_rating
```

Comment:
Driver and Passenger methods should handle incomplete trips.
Both these methods will hit an error if you try to run them on a user with an incomplete trip. There are a couple of workarounds for this.

You could explicitly ignore trips with a `nil` cost:

```ruby
trips.each do |trip|
  if trip.cost.nil?
    next
  end
  # ... add to the total ...
end
```

Or even better, you could write a helper method that returns a list of complete trips:

```ruby
def completed_trips
  return @trips.reject { |t| t.end_time.nil? }
end

def net_expenditures
  completed_trips.each do |trip|
    # ... same logic as before ...
  end
end
```